src/GitHub.App/Controllers/NavigationController.cs

```diff
@@ -66,7 +67,16 @@ public void LoadView(IConnection connection, ViewWithData data, Action<IView> on
             switch (data.MainFlow)
             {
                 case UIControllerFlow.PullRequestCreation:
+                    //The selected flow will change after the view is created, so save it.
+                    var previousFlow = Current?.SelectedFlow;
+
                     CreateView(connection, data, onViewLoad);
+                    if (data.Data == null && previousFlow == UIControllerFlow.PullRequestCreation && history.Count >= 2)
+                    {
+                        Debug.Assert(history.Count >= 2, "History shouldn't have less than 2 items in it when visiting the creation view.");
```

Comment:
You're already testing `history.Count >= 2` in the `if` statement so this `Debug.Assert` is unnecessary.